src/index.ts

```diff
@@ -33,7 +33,7 @@ function pagesPlugin(userOptions: UserOptions = {}): Plugin {
       ctx.setupViteServer(server)
     },
     resolveId(id) {
-      if (MODULE_IDS.includes(id))
+      if (id === userOptions.moduleId || MODULE_IDS.includes(id))
```

Comment:
This will not works, because it will resolved to the same virtual id.
For multiple instance, we need different virtual id.